tests/consumer.rs

```diff
@@ -103,3 +95,198 @@ async fn test_stream_consumer_offset_out_of_range() {
     // stream ends
     assert!(stream.next().await.is_none());
 }
+
+#[tokio::test]
+async fn test_stream_consumer_start_at_earliest() {
```

Comment:
Do we want a test that manually truncates the topic, so that the earliest offset is guaranteed to not be 0?

Reply:
good idea, I'll add a test (that again only runs w/ Kafka since redpanda doesn't support deletes).

Reply:
done